pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -335,6 +346,27 @@ public void healthCheck(@Suspended AsyncResponse asyncResponse,
                 });
     }
 
+    private void checkDeadlockedThreads() {
```

Comment:
Are we able to add unit test for this?

Reply:
+1

Reply:
A unit test would be nice to add, but if it's hard to implement, I think we should still merge this PR.

Reply:
> A unit test would be nice to add, but if it's hard to implement, 
Constructing a dead lock should not be very hard. Although I am not sure what side effect would it bring to CI procedure.

> I think we should still merge this PR.
Not blocking this. :)

Reply:
@lhotari Can this method identify the deadlocks we encountered in `metadata-thread` before?

Reply:
@Jason918 If it's a clear deadlock based on a lock or an object monitor, it will be detected.

Reply:
I was able to add a unit test `org.apache.pulsar.broker.admin.AdminApiHealthCheckTest#testHealthCheckupDetectsDeadlock`.

It demonstrates what gets logged when a deadlock is detected.

Here's some of the logging:
```
Found one Java-level deadlock:
=============================
"deadlockthread-1":
  waiting for ownable synchronizer 0x00000000d896b550, (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync),
  which is held by "deadlockthread-2"
"deadlockthread-2":
  waiting for ownable synchronizer 0x00000000d896b4d8, (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync),
  which is held by "deadlockthread-1"

Java stack information for the threads listed above:
===================================================
"deadlockthread-1":
	at jdk.internal.misc.Unsafe.park(java.base@11.0.13/Native Method)
	- parking to wait for  <0x00000000d896b550> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.13/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@11.0.13/AbstractQueuedSynchronizer.java:885)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(java.base@11.0.13/AbstractQueuedSynchronizer.java:917)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@11.0.13/AbstractQueuedSynchronizer.java:1240)
	at java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.lock(java.base@11.0.13/ReentrantReadWriteLock.java:959)
	at org.apache.pulsar.broker.admin.AdminApiHealthCheckTest.deadlock(AdminApiHealthCheckTest.java:133)
	at org.apache.pulsar.broker.admin.AdminApiHealthCheckTest.lambda$testHealthCheckupDetectsDeadlock$4(AdminApiHealthCheckTest.java:109)
	at org.apache.pulsar.broker.admin.AdminApiHealthCheckTest$$Lambda$720/0x00000001008abc40.run(Unknown Source)
	at java.lang.Thread.run(java.base@11.0.13/Thread.java:829)
"deadlockthread-2":
	at jdk.internal.misc.Unsafe.park(java.base@11.0.13/Native Method)
	- parking to wait for  <0x00000000d896b4d8> (a java.util.concurrent.locks.ReentrantReadWriteLock$NonfairSync)
	at java.util.concurrent.locks.LockSupport.park(java.base@11.0.13/LockSupport.java:194)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(java.base@11.0.13/AbstractQueuedSynchronizer.java:885)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(java.base@11.0.13/AbstractQueuedSynchronizer.java:917)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(java.base@11.0.13/AbstractQueuedSynchronizer.java:1240)
	at java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.lock(java.base@11.0.13/ReentrantReadWriteLock.java:959)
	at org.apache.pulsar.broker.admin.AdminApiHealthCheckTest.deadlock(AdminApiHealthCheckTest.java:133)
	at org.apache.pulsar.broker.admin.AdminApiHealthCheckTest.lambda$testHealthCheckupDetectsDeadlock$5(AdminApiHealthCheckTest.java:113)
	at org.apache.pulsar.broker.admin.AdminApiHealthCheckTest$$Lambda$721/0x00000001008ab040.run(Unknown Source)
	at java.lang.Thread.run(java.base@11.0.13/Thread.java:829)

Found 1 deadlock.
```

Reply:
I have added a unit test